x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -579,13 +585,41 @@ export const updateInstallStatus = async ({
   });
 };
 
+export async function restartInstallation(options: {
+  savedObjectsClient: SavedObjectsClientContract;
+  pkgName: string;
+  pkgVersion: string;
+  installSource: InstallSource;
+  verificationResult?: PackageVerificationResult;
+}) {
+  const { savedObjectsClient, pkgVersion, pkgName, installSource, verificationResult } = options;
+
+  let savedObjectUpdate: Partial<Installation> = {
+    install_version: pkgVersion,
+    install_status: 'installing',
+    install_started_at: new Date().toISOString(),
+    install_source: installSource,
+  };
+
+  if (verificationResult) {
+    savedObjectUpdate = {
+      ...savedObjectUpdate,
+      verification_key_id: null, // unset any previous verification key id
```

Comment:
Not sure if this is a bit messy, if for example I install a packge and it is verified, then I reinstall it and it is "unknown", we need to unset the verification key ID, I've done this by nulling it.

Reply:
This makes sense to me